Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -41,6 +43,10 @@ static PropertiesConfiguration getConfig(String configFile) {
                     AbstractConfiguration.setDefaultListDelimiter('&');
                     try {
                         initialized = new PropertiesConfiguration(configFile);
+                        if (logger.isDebugEnabled()) {
```

Comment:
Why check for this? just use logger.debug and the user can decide, which level he wants to see on his log

Reply:
Because the line is a short form of the following:

    FileSystem fileSystem = FileSystem.getDefaultFileSystem();
    URL url = ConfigurationUtils.locate(fileSystem, null, configFile);
    StringBuilder stringBuilder = new StringBuilder();
    stringBuilder.append("Using configuration from ");
    stringBuilder.append(url);
    String message = stringBuilder.toString();
    logger.debug(message);

All of these actions are unnecessary, if debug level is disabled, and should therefore be skipped in that case.

Reply:
Sorry I don't quite understand your answer.
why are you checking for `isDebugEnabled` ?
I don't think we do this at any other place in the code.
Just add `logger.debug` and everything is fine.

Reply:
The thing is, that Java invokes the commands in a specific order:

                            FileSystem.getDefaultFileSystem() 
                                                 ↓
                      ConfigurationUtils.locate(..., null, configFile)
                                   ↓
    "Using configuration from " + ...
                  ↓
    logger.debug(...);

This means, if debug level is **disabled**, Java _will still_ load the file system, resolve the absolute path of the file on disk, and build a log string, before passing it to `logger` which decides that it doesn’t need to log it. Therefore, if retrieving the information to log is “work” (not just passing existing variables) it should be checked **before**, if it shall be logged at all in the end, or not.

There **are** some other places where these checks can be found in the code, especially for the “trace” level.

- https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/helper/metadata/legacytypeimplementations/LegacyLogicalDocStructHelper.java#L252-L257
- https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java#L295-L298
- …